Framework/src/TObject2JsonServer.cxx

```diff
@@ -1,33 +1,89 @@
+
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
 ///
-/// \file   TObject2JsonServer.cxx
+/// \file   TObejct2Json.cxx
+/// \author Vladimir Kosmala
 /// \author Adam Wegrzynek
 ///
 
 // TObject2Json
-#include "TObject2JsonFactory.h"
-#include <boost/program_options.hpp>
+#include "TObject2JsonServer.h"
+#include "TObject2JsonWorker.h"
+#include "TObject2JsonBackendFactory.h"
+#include "QualityControl/QcInfoLogger.h"
+
+// ZMQ
+#include "zmq.h"
+
+// Boost
+#include <boost/algorithm/string.hpp>
+
+#include <TROOT.h>
+
+using namespace std;
+using namespace o2::quality_control::core;
+using namespace std::string_literals;
+using o2::quality_control::tobject_to_json::TObject2JsonBackendFactory;
 
-using o2::quality_control::tobject_to_json::TObject2JsonFactory;
+namespace o2 {
+namespace quality_control {
+namespace tobject_to_json {
 
-int main(int argc, char *argv[])
+TObject2JsonServer::TObject2JsonServer()
+  : mCtx(1),
+    mFrontend(mCtx, ZMQ_ROUTER),
+    mBackend(mCtx, ZMQ_DEALER)
 {
-  boost::program_options::variables_map vm;
-  boost::program_options::options_description desc("Allowed options");
-  desc.add_options()
-    ("backend", boost::program_options::value<std::string>()->required(), "Backend URL, eg.: mysql://<loign>:<password>@<hostname>:<port>/<database>")
-    ("zeromq-server", boost::program_options::value<std::string>()->required(), "ZeroMQ server endpoint, eg.: tcp://<host>:<port>")
-  ;
-  try {
-    boost::program_options::store(boost::program_options::parse_command_line(argc, argv, desc), vm);
-    boost::program_options::notify(vm);
-  } catch (...) {
-    std::cout << desc << std::endl;
-    return 1;
+  // Make ROOT aware that workers will do calls from threads
+  ROOT::EnableThreadSafety();
+}
+
+void TObject2JsonServer::start(std::string backendUrl, std::string zeromq, uint8_t numThreads)
+{
+  if (numThreads <= 0) {
+    throw std::runtime_error("Number of workers must be >= 1");
   }
-  std::string backend = vm["backend"].as<std::string>();
-  std::string zeromq = vm["zeromq-server"].as<std::string>();
 
-  auto converter = TObject2JsonFactory::Get(backend, zeromq);
-  converter->start();
-  return 0;
+  std::vector<std::unique_ptr<TObject2JsonWorker>> workers;
+
+  QcInfoLogger::GetInstance() << "Deploying workers..." << infologger::endm;
+  for (int i = 0; i < numThreads; ++i) {
+    std::unique_ptr<Backend> backendInstance = TObject2JsonBackendFactory::get(backendUrl);
+    TObject2JsonWorker *worker = new TObject2JsonWorker(mCtx, std::move(backendInstance));
```

Comment:
Is it possible to use `make_unique` instead of getting raw pointer and then converting it into unique?
`auto worker = std::make_unique<TObject2JsonWorker>(mCtx, std::move(backendInstance));`